### PR TITLE
Replace specialized parameter deprecation code with a general utility

### PR DESCRIPTION
**Pull request by kchawla-pi**
This PR intends to undo the code redundancy introduced via PRs #1913 & #1918 by refactoring the functionality into one utility function. This will also help with the same functionality in Nistats and address code redundancy there as well.
Finally I foresee this as a useful function in future use cases as well.

### CODE REVIEW

File: nilearn/tests/test_helpers.py [hunk 1]
```diff
@@ -0,0 +1,127 @@
...
+def test_version_warn_deprecated_params():
+    mock_input, replacement_params = _mock_args_for_testing_replace_parameter()
+    expected_warnings = [
+        ('The parameter "deprecated_param_0" will be removed in Nilearn version 0.6.1rc. Please use the parameter "replacement_param_0" instead.'),
+        ('The parameter "deprecated_param_1" will be removed in Nilearn version 0.6.1rc. Please use the parameter "replacement_param_1" instead.'),
+        ]
+    with warnings.catch_warnings(record=True) as raised_warnings:
+        helpers._warn_deprecated_params(
+                replacement_params,
+                end_version='0.6.1rc',
```
Comment (bthirion) [inline nilearn/tests/test_helpers.py]: oops; we'll have to update this test for all releases ?

File: nilearn/tests/test_helpers.py [hunk 2]
```diff
@@ -0,0 +1,127 @@
...
+                lib_name='other_lib',
+                kwargs=mock_input,
+                )
+    for raised_warning_, expected_warning_ in zip(raised_warnings,
+                                                  expected_warnings):
+        assert str(raised_warning_.message) == expected_warning_
+
+
+def test_replace_parameters():
+    mock_input, replacement_params = _mock_args_for_testing_replace_parameter()
```
Comment (bthirion) [inline nilearn/tests/test_helpers.py]: Could you please add some comments to explain what the tests are probing ? I find them quite abstract/hard to follow.

File: nilearn/tests/test_helpers.py (hunk 1 above)
Comment (kchawla-pi) [reply nilearn/tests/test_helpers.py]: No, it's merely checking that the end version is being displayed properly, the version I chose is arbitrary.

**Comment by kchawla-pi** [comment]
@bthirion Check it out.
